lib/vtul/dataingest.rb

```diff
@@ -0,0 +1,184 @@
+require 'csv'
+require 'digest'
+require 'fileutils'
+require 'json'
+require 'net/http'
+require 'uri'
+require 'yaml'
+
+class DataIngest
+    
+  def createreport(filename, content)
+    File.open(filename, 'w') { |file| file.write(content) }
+  end
+
+
+  def getfilelist(folderpath)
+    return Dir[folderpath].select{ |filename| File.file? filename }.map{ |filename| File.basename filename }
+  end
+
+
+  def getfileprefix(filename)
+    if filename.start_with?("dmf_")
+      return "dmf_"
+    elsif filename.start_with?("cgit_")
+      return "cgit_"
+    else
+      return ""
+    end
+  end
+
+
+  def geojson(content)
+    rec = {}
+    uuid = Digest::SHA256.hexdigest content[1] 
+    timestring = Time.now().strftime("%Y-%m-%dT%H:%M:%SZ") 
+
+    rec['uuid'] = uuid
+    rec['dc_identifier_s'] = content[1]
+    rec['dc_rights_s'] = content[2]
+    rec['dct_provenance_s'] = content[3]
+    if !content[4].nil?
+      rec['dct_references_s'] = "{\"http://schema.org/downloadUrl\":\"" + content[1] + "\"}"
+    else
+      rec['dct_references_s'] = "{\"http://schema.org/downloadUrl\":\"" + content[1] + "\",\"http://www.opengis.net/def/serviceType/ogc/wcs\":\"" + content[4] + "\"}"     
+    end
+    rec['dc_creator_sm'] = content[5]
+    rec['dc_language_s'] = content[6]
+    rec['dc_publisher_s'] = content[7]
+    rec['dc_type_s'] = content[9]
+    rec['dct_spatial_sm'] = content[10]
+    rec['dct_temporal_sm'] = content[11]
+    rec['dct_issued_s'] = timestring
+    rec['dct_isPartOf_sm'] = content[13]
+    rec['georss_box_s'] = content[14]
+    if !content[14].nil?
+      gdata = content[14].split(",")  
+      rec['solr_geom'] = "ENVELOPE("+ gdata[1] + "," + gdata[3] + "," + gdata[2] + "," + gdata[0] + ")"
+    end
+    rec['dc_title_s'] = content[17]
+    rec['dc_description_s'] = content[18]
+    rec['dc_format_s'] = content[19]
+    rec['dc_subject_sm'] = content[20]
+    rec['layer_id_s'] = content[21]
+    rec['layer_modified_dt'] = timestring
+    rec['layer_slug_s'] = content[23]
+    rec['layer_geom_type_s'] = content[24]
+
+    return rec.to_json
+  end
+
+
+  def ingestrec(solrurl, content)
```

Comment:
Does this function output some kind of logs for us to parse and generate reports for Shane?